### DISCUSSION TITLE
movies_embeddings table in Manticore

### DISCUSSION THREAD

**Discussion by sarafarahabadi** [sovelluskontti/project-semantic-search#48] [General]
Hello Turo, 

I have been working on the upload but no luck so far. I have a Python script to insert each row, but getting the data from the CSV file has been challenging. Below is the sample:

# cursor = conn.cursor()

# id = 1
# title = 'testing '
# embedding = "(-0.0641787201166153, ... , -0.054265085607767105)"

# sql = f"INSERT INTO movies (id, title, embedding) VALUES ({id}, '{title}', {embedding})"

# cursor.execute(sql)

# conn.commit()

# cursor.close()
# conn.close()

# print("Data inserted successfully.")

I was checking and I saw that there is a movies_embeddings table in the database which has the embeddings. 
I want to know how you could do it and thanks for the help. 

Br,
Sara

**Comment by sarafarahabadi** [comment]
Also now I saw the the table's structure is like this:
mysql> describe movies_embeddings;
+-----------+--------+----------------+
| Field     | Type   | Properties     |
+-----------+--------+----------------+
| id        | bigint |                |
| title     | text   | indexed stored |
| embedding | text   | indexed stored

**Reply by nyluntu** [reply]
I will try to check this today.

**Reply by sarafarahabadi** [reply]
I am working on a solution that I hope it will work. I will keep you updated if it does.

**Reply by nyluntu** [reply]
https://github.com/sovelluskontti/project-semantic-search/pull/50/files

Check the load_data.py

it's the most important part. No need to worry about other changed files. I try to send video explain tomorrow, now it just too late. :)

Before trying the load_data.py, use create table command from sql-examples.sql 

The issue with your table schema was caused by manticore auto schema feature. If the table not exist during the inserts, it will auto create them. That's why the manticore just try to figure out right data type that was in string format. Creating the table before inserting the data is better choise.

**Reply by sarafarahabadi** [reply]
Hello Turo, 
Thank you so much for the help. It works perfectly now. I cleaned the code a little and the main branch is fully functioning both for full-text search and semantic search when putting SQL query with embeddings.

**Reply by nyluntu** [reply]
@sarafarahabadi I uploaded a new video to shared files. Check it. Link to the tty thing: https://kossy0701.medium.com/what-is-tty-true-in-docker-compose-yml-47a72891aee2

See you on Monday!
